src/interchange/src/avro/decode.rs

```diff
@@ -91,13 +91,16 @@ impl Decoder {
     }
 
     /// Decodes Avro-encoded `bytes` into a `Row`.
-    pub async fn decode(&mut self, bytes: &mut &[u8]) -> anyhow::Result<Row> {
+    pub async fn decode(&mut self, bytes: &mut &[u8]) -> Result<Result<Row, Error>, Error> {
         // Clear out any bytes that might be left over from
         // an earlier run. This can happen if the
         // `dsr.deserialize` call returns an error,
         // causing us to return early.
         let mut packer = self.row_buf.packer();
-        let (bytes2, resolved_schema, csr_schema_id) = self.csr_avro.resolve(bytes).await?;
+        let (bytes2, resolved_schema, csr_schema_id) = match self.csr_avro.resolve(bytes).await? {
```

Comment:
the `?` should be documented as handling transient errors

Reply:
What we actually need to do is define a `TransientError` and various `OtherError` types so that `?` doesn't accidentally work with both, but that's something for a future PR

Reply:
sounds good!